Mobile/ShopLens/Droid/Activities/CameraActivity.cs

```diff
@@ -218,11 +222,23 @@ private void RecogniseImage(Uri uri)
                         QueueMode.Flush,
                         null,
                         null);
-
+                    new ErrorDialogCreator(this, "Shopping cart", "Would you like to add to thid product to your shopping cart?", "Yes", "No", 
+                        addToShoppingCart, doNotAddToShoppingCart);
```

Comment:
No's Action should be null, because why would you ever call an empty method? Unless, as I've said, that it is important and it will do something useful.